opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/PlatformResources.java

```diff
@@ -33,113 +35,380 @@
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
+import decodes.db.DatabaseException;
+import decodes.db.DatabaseIO;
+import decodes.db.Platform;
+import decodes.db.PlatformConfig;
+import decodes.db.PlatformList;
+import decodes.db.PlatformSensor;
+import decodes.db.PlatformStatus;
+import decodes.db.Site;
+import decodes.db.TransportMedium;
+import decodes.sql.DbKey;
+import decodes.tsdb.DbIoException;
+import opendcs.dai.PlatformStatusDAI;
 import org.opendcs.odcsapi.beans.ApiPlatform;
 import org.opendcs.odcsapi.beans.ApiPlatformRef;
-import org.opendcs.odcsapi.dao.ApiPlatformDAO;
+import org.opendcs.odcsapi.beans.ApiPlatformSensor;
+import org.opendcs.odcsapi.beans.ApiPlatformStatus;
+import org.opendcs.odcsapi.beans.ApiTransportMedium;
 import org.opendcs.odcsapi.dao.DbException;
-import org.opendcs.odcsapi.errorhandling.ErrorCodes;
 import org.opendcs.odcsapi.errorhandling.WebAppException;
-import org.opendcs.odcsapi.hydrojson.DbInterface;
 import org.opendcs.odcsapi.sec.AuthorizationCheck;
-import org.opendcs.odcsapi.util.ApiConstants;
-import org.opendcs.odcsapi.util.ApiHttpUtil;
 
 @Path("/")
-public class PlatformResources
+public class PlatformResources extends OpenDcsResource
 {
 	@Context HttpHeaders httpHeaders;
 
 	@GET
 	@Path("platformrefs")
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
-	public Response gePlatformRefs(@QueryParam("tmtype") String tmtype)
-		throws DbException, SQLException
+	public Response getPlatformRefs(@QueryParam("tmtype") String tmtype)
+			throws DbException
 	{
 		HashMap<String, ApiPlatformRef> ret = new HashMap<>();
-		Logger.getLogger(ApiConstants.loggerName).fine("getPlatforms, tmtype=" + tmtype);
-		try (DbInterface dbi = new DbInterface();
-			ApiPlatformDAO platformDAO = new ApiPlatformDAO(dbi))
+		try
 		{
-			ArrayList<ApiPlatformRef> platSpecs = platformDAO.getPlatformRefs(tmtype);
+			// TODO: Add support for filtering by transport media type
+			DatabaseIO dbio = getLegacyDatabase();
+			PlatformList platformList = new PlatformList();
+			dbio.readPlatformList(platformList);
+			ArrayList<ApiPlatformRef> platSpecs = map(platformList);
 			for(ApiPlatformRef ps : platSpecs)
 				ret.put(ps.getName(), ps);
+			return Response.status(HttpServletResponse.SC_OK).entity(ret).build();
+		}
+		catch (DatabaseException ex)
+		{
+			throw new DbException("Unable to retrieve platform list", ex);
 		}
-		Logger.getLogger(ApiConstants.loggerName).fine("getPlatforms returning " + ret.size() + " objects.");
-		return ApiHttpUtil.createResponse(ret);
 	}
-	
+
+	static ArrayList<ApiPlatformRef> map(PlatformList platformList)
+	{
+		ArrayList<ApiPlatformRef> ret = new ArrayList<>();
+		Iterator<Platform> platform = platformList.iterator();
+		while (platform.hasNext())
+		{
+			ApiPlatformRef ref = new ApiPlatformRef();
+			Platform plat = platform.next();
+			ref.setName(plat.getDisplayName());
```

Comment:
Can this call the single object map() method?

Reply:
Since the ApiPlatformRef has a different structure to ApiPlatform, the map can't be reused.